src/platform/silabs/tracing/SilabsPowerTracing.cpp

```diff
@@ -0,0 +1,184 @@
+                                         SL_POWER_MANAGER_EVENT_TRANSITION_ENTERING_EM2),
+                                        StaticPowerManagerTransitionCallback },
+    mStatisticsTimer(nullptr), mInitialized(false)
+{}
+
+CHIP_ERROR SilabsPowerTracing::Init()
+{
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    bool powerManagerSubscribed = false;
+
+    // Return early if already initialized
+    if (mInitialized)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    // Allocate energy trace storage
+    if (mEnergyTraces == nullptr)
+    {
+        mEnergyTraces = static_cast<EnergyTrace *>(calloc(SILABS_TRACING_ENERGY_TRACES_MAX, sizeof(EnergyTrace)));
+        if (mEnergyTraces == nullptr)
+        {
+            err = CHIP_ERROR_NO_MEMORY;
+        }
+    }
+
+    if (err == CHIP_NO_ERROR)
+    {
+        mEnergyTraceCount = 0;
+
+        // Initialize power manager and subscribe to events
+        sl_power_manager_init();
+        sl_power_manager_subscribe_em_transition_event(&mPowerManagerEmTransitionEventHandle, &mPowerManagerEmTransitionEventInfo);
+        powerManagerSubscribed = true;
+
+        // Create and start one-shot timer for statistics output
+        mStatisticsTimer = osTimerNew(OnPowerManagerStatisticsTimer, osTimerOnce, nullptr, nullptr);
+        if (mStatisticsTimer == nullptr)
+        {
+            ChipLogError(DeviceLayer, "Failed to create power manager statistics timer");
+            err = CHIP_ERROR_NO_MEMORY;
+        }
+    }
+
+    if (err == CHIP_NO_ERROR)
+    {
+        uint32_t ticks = SILABS_TRACING_ENERGY_TRACES_SECONDS * osKernelGetTickFreq();
+        if (osTimerStart(mStatisticsTimer, ticks) != osOK)
+        {
+            ChipLogError(DeviceLayer, "Failed to start power manager statistics timer");
+            err = CHIP_ERROR_INTERNAL;
+        }
+    }
+
+    // Cleanup on error
+    if (err != CHIP_NO_ERROR)
+    {
+        if (mStatisticsTimer != nullptr)
+        {
+            osTimerDelete(mStatisticsTimer);
+            mStatisticsTimer = nullptr;
+        }
+
+        if (powerManagerSubscribed)
+        {
+            sl_power_manager_unsubscribe_em_transition_event(&mPowerManagerEmTransitionEventHandle);
+        }
+
+        if (mEnergyTraces != nullptr)
+        {
+            free(mEnergyTraces);
+            mEnergyTraces     = nullptr;
+            mEnergyTraceCount = 0;
+        }
+    }
+    else
+    {
+        mInitialized = true;
+    }
+
+    return err;
+}
+
+SilabsPowerTracing::~SilabsPowerTracing()
+{
+    if (mStatisticsTimer != nullptr)
+    {
+        osTimerStop(mStatisticsTimer);
+        osTimerDelete(mStatisticsTimer);
+        mStatisticsTimer = nullptr;
+    }
+
+    if (mEnergyTraces != nullptr)
+    {
+        free(mEnergyTraces);
+        mEnergyTraces     = nullptr;
+        mEnergyTraceCount = 0;
+    }
+
+    sl_power_manager_unsubscribe_em_transition_event(&mPowerManagerEmTransitionEventHandle);
```

Comment:
Since our subscribe call `sl_power_manager_subscribe_em_transition_event ` is done outside of this object, this probably shouldn't be here, what happens if this is called and we are not subscribed?

Could the object instead handle both the register and unregister?